tests/test_mappings.py

```diff
@@ -0,0 +1,81 @@
+from hapi.pipelines.utilities.mappings import get_code_from_name
+
+
+def test_get_code_from_name():
+    sector_lookup = {
+        "Emergency Shelter and NFI": "SHL",
+        "Camp Coordination / Management": "CCM",
+        "Mine Action": "PRO - MIN",
+        "Food Security": "FSC",
+        "Water Sanitation Hygiene": "WSH",
+        "Logistics": "LOG",
+        "Child Protection": "PRO - CPN",
+        "Protection": "PRO",
+        "Education": "EDU",
+        "Nutrition": "NUT",
+        "Health": "HEA",
+        "Early Recovery": "ERY",
+        "Emergency Telecommunications": "TEL",
+        "Gender Based Violence": "PRO - GBV",
+        "Housing, Land and Property": "PRO - HLP",
+    }
+    sector_map = {
+        "abris": "SHL",
+        "cccm": "CCM",
+        "coordination": "CCM",
+        "education": "EDU",
+        "eha": "WSH",
+        "erl": "ERY",
+        "nutrition": "NUT",
+        "operatioanl presence: water, sanitation & hygiene": "WSH",
+        "operational presence: education in emergencies": "EDU",
+        "operational presence: emergency shelter & non-food items": "SHL",
+        "operational presence: food security & agriculture": "FSC",
+        "operational presence: health": "HEA",
+        "operational presence: nutrition": "NUT",
+        "operational presence: protection": "PRO",
+        "protection": "PRO",
+        "sa": "FSC",
+        "sante": "HEA",
+        "wash": "WSH",
+    }
+
+    org_type_lookup = {
+        "Academic / Research": "431",
+        "Donor": "433",
+        "Embassy": "434",
+        "Government": "435",
+        "International NGO": "437",
+        "International Organization": "438",
+        "Media": "439",
+        "Military": "440",
+        "National NGO": "441",
```

Comment:
What happens if you look for "NGO"? Does it fail or match to one of the two with NGO? Similarly, what happens if you put "International"?

Reply:
Hm it returns None for International, but National NGO for NGO...  it's pretty strict on matching the number of words, but is a little looser when it comes to something close like "ingo" vs "nngo".  I could allow fuzzy matching only for the official list of categories - right now I'm checking those first but if I don't find a match I'm checking against the other mappings in the yaml.